src/tls.c

```diff
@@ -6222,6 +6222,10 @@ static int TLSX_SupportedVersions_Parse(WOLFSSL* ssl, const byte* input,
             /* Downgrade the version. */
             ssl->version.minor = minor;
         }
+
+        if (ssl->options.downgrade && (minor < ssl->options.minDowngrade)) {
```

Comment:
ssl->options.downgrade indicates whether downgrade is allowed.
We want to be checking wether downgrading is happening.
What are we attempting to catch?
ssl->version.minor = ?
minor = ?
ssl->options.minDowngrade = ?
(Values at line 6215)

Reply:
Hi @SparkiDev,

Reworking #3983. 
The goal is to get the following configurations to behave as follows: 

fail:
- a server configured with WOLFSSL_TLSV1_2
- a client configured with WOLFSSL_TLSV1_2 and WOLFSSL_TLSV1_3 but minDowngrade is set to WOLFSSL_TLSV1_3
 
pass:
- a server configured with WOLFSSL_TLSV1_3
- a client configured with WOLFSSL_TLSV1_3 but minDowngrade is set to WOLFSSL_TLSV1_3

Reply:
So FAIL (in client):
ssl->version.minor = TLS v1.2
minor = TLS v1.2 (from server)
ssl->options.minDowngrade = TLS v1.3

PASS (in client):
ssl->version.minor = TLS v1.3
minor = TLS v1.3 (from server)
ssl->options.minDowngrade = TLS v1.3

Only problem with this is that TLS v1.2 handshakes on the server aren't meant to parse the SupportedVersions extension and aren't meant to respond with a KeyShare extension.
In the fail case, is the server configured for TLS v1.2 and capable of TLS v1.3?

If the client has min downgrade of TLS v1.3 then it should not have sent TLS v1.2 in the SupportedVersions extension in the first place. That would be a good thing to fix.

Are we controlling both sides in User's test case that was fixed?

Reply:
>If the client has min downgrade of TLS v1.3 then it should not have sent TLS v1.2 in the SupportedVersions extension in the first place. That would be a good thing to fix.

Sounds like, we should fix this first. After the fix, server won't accept ClientHello if server doesn't have capable of TLS1.3. 

@tmael 
Do you think your test case will be satisfied with the fix above? I think Server will become failure instead of Client.

Reply:
@miyazakh, @SparkiDev, At the moment I've no further comments on this PR since it's generating more questions than answers.

Reply:
Hi @tmael 
Ok, that is fine. I will try to implement the fix that Sean suggests.